featuretools/feature_base/feature_base.py

```diff
@@ -64,9 +64,9 @@ def copy(self):
         raise NotImplementedError("Must define copy on FeatureBase subclass")
 
     def get_name(self):
-        if self._name is not None:
-            return self._name
-        return self.generate_name()
+        if not self._name:
```

Comment:
@rwedge I realized that `_name` is defined as a class variable, but when we generate the name we then create an instance variable shadowing the class variable. This seems more confusing to me than putting `self._name = None` in the initializer, or is it a python idiom that I'm not familiar with?

Reply:
you're right. we should move to `self._name = None` in the init